benchmarks/helics/WattsStrogatzFederate.hpp

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright (c) 2017-2020,
+Battelle Memorial Institute; Lawrence Livermore National Security, LLC; Alliance for Sustainable
+Energy, LLC.  See the top-level NOTICE for additional details. All rights reserved.
+SPDX-License-Identifier: BSD-3-Clause
+*/
+#pragma once
+
+#include "BenchmarkFederate.hpp"
+#include "helics/application_api/Endpoints.hpp"
+#include "helics/application_api/MessageFederate.hpp"
+#include "helics/core/ActionMessage.hpp"
+
+#include <algorithm>
+#include <random>
+#include <string>
+#include <utility>
+#include <vector>
+
+/** class implementing a Watts-Strogatz-like communication pattern with messages*/
+class WattsStrogatzFederate: public BenchmarkFederate {
+  public:
+    int initialMessageCount{
+        20};  // number of messages the federate should send when it starts (too high results in
+              // lower degrees being slower than high, possibly queue or buffer related for a
+              // socket, and can make the udp benchmark hang indefinitely)
+
+    // A typical Watts-Strogatz graph uses degree K for total connections to neighbors, K/2 on each
+    // side. Since this requires more coordination to setup than is easily done from within
+    // Federates, this simplified version has each node setup the connectiosn on its right side only
+    // (so k=K/2 in this benchmark)
+    int k{1};  // degree
+    double b{0};  // re-wire probability for each of k rightmost neighbors
+
+    // Classes related to the exponential and uniform distribution random number generator
+    bool generateRandomSeed{false};
+    unsigned int seed{0xABad5eed};  // suggestions for seed choice were not having a majority of the
+                                    // bits as 0 is better
+    std::mt19937 rand_gen;
```

Comment:
how much variation is there depending on the seed.  It looks like in the default case you have the seed the same for each case, which could make the test skewed in some fashion.

Reply:
The variation as a result of other factors in test runs seemed to be larger than variation due to changing the seed.